util.go

```diff
@@ -180,3 +181,43 @@ func fmtifaces(v []interface{}) string {
 
 	return str.String()
 }
+
+// Encode index slice into single string
+func encodeIndices(indices []int) string {
+	var sb strings.Builder
+	for _, index := range indices {
+		sb.WriteString("/")
+		sb.WriteString(strconv.Itoa(index))
+	}
+	return sb.String()
+}
+
+// Get field by index and init pointers on path if flag is true
```

Comment:
let's add some credit to original source here
something like: `modified from: https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/reflect/value.go;l=1228-1245;bpv=1;bpt=1`